structure/fenwicktree/fenwicktree.go

```diff
@@ -0,0 +1,62 @@
+//Fenwick Tree Data Structure for Range Queries. Also known as Binary Indexed Tree
+//Build: O(N)
+//Query: O(log(N))
+//Update: O(log(N))
+//reference: https://brilliant.org/wiki/fenwick-tree/
+
+package fenwicktree
+
+// n: Size of the input array.
+// array: the input array on which queries are made.
+// bit: store the sum of ranges.
+type FenwickTree struct {
+	n     int
+	array []int
+	bit   []int
+}
+
+// This function creates a new Fenwick tree, initialized array and bit with
+// the values of the array. Note that the queries and updates should have
+// one based indexing.
+func NewFenwickTree(array []int) *FenwickTree {
+	n := len(array)
```

Comment:
You are only using this value once, so why the variable?

Reply:
Okay, I'll remove it